Please, add support for `eslint` annotations

[ESLint](https://eslint.org/) is able to generate reports in `json` format, e.g.:
```sh
yarn run eslint -f json -o reports/eslint.json app/javascript/controllers/index.js
```
Here is an example [eslint.json](https://github.com/cirruslabs/cirrus-ci-annotations/files/7409849/eslint.json.gz) from the `reports` folder.

The example file to test is `app/javascript/controllers/index.js`:
```js
// This file is auto-generated by ./bin/rails stimulus:manifest:update
// Run that command whenever you add a new controller

import {application} from "./application";

import SearchableController from "./searchable_controller";

import BackSearchableController from "./back_searchable_controller";

import SortableController from "./sortable_controller";

import TestController from "./test_controller";

application.register("searchable", SearchableController);
application.register("back-searchable", BackSearchableController);
application.register("sortable", SortableController);
application.register("test", TestController);
```

Its corresponding plain text report looks like this:

```
  4:9  error  Replace `application` with `·application·`  prettier/prettier

✖ 1 problem (1 error, 0 warnings)
  1 error and 0 warnings potentially fixable with the `--fix` option.
```

--- a/annotations.go
+++ b/annotations.go
@@ -13,6 +13,8 @@ func ParseAnnotations(format string, path string) (error, []model.Annotation) {
 	switch strings.ToLower(format) {
 	case "junit":
 		return parsers.ParseJUnitAnnotations(path)
+	case "eslint":
+		return parsers.ParseESLintAnnotations(path)
 	case "golangci":
 		return parsers.ParseGoLangCIAnnotations(path)
 	case "android-lint":

--- a/parsers/eslint.go
+++ b/parsers/eslint.go
@@ -1,0 +1,64 @@
+package parsers
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/cirruslabs/cirrus-ci-annotations/model"
+	"io/ioutil"
+)
+
+type eslintMessage struct {
+	RuleID   string `json:"ruleId"`
+	Severity int    `json:"severity"`
+	Message  string `json:"message"`
+	Line     int64  `json:"line"`
+	Column   int64  `json:"column"`
+	NodeType string `json:"nodeType"`
+}
+
+type eslintResult struct {
+	FilePath string          `json:"filePath"`
+	Messages []eslintMessage `json:"messages"`
+}
+
+func eslintSeverityToAnnotationLevel(eslintSeverity int) model.AnnotationLevel {
+	switch eslintSeverity {
+	case 1:
+		return model.LevelWarning
+	case 2:
+		return model.LevelFailure
+	default:
+		return model.LevelNotice
+	}
+}
+
+func ParseESLintAnnotations(path string) (error, []model.Annotation) {
+	var result []model.Annotation
+
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err, nil
+	}
+
+	var eslintResults []eslintResult
+	err = json.Unmarshal(data, &eslintResults)
+	if err != nil {
+		return err, nil
+	}
+
+	for _, eslintResult := range eslintResults {
+		for _, eslintMessage := range eslintResult.Messages {
+			result = append(result, model.Annotation{
+				Level:       eslintSeverityToAnnotationLevel(eslintMessage.Severity),
+				Message:     fmt.Sprintf("%s: %s", eslintMessage.RuleID, eslintMessage.Message),
+				Path:        eslintResult.FilePath,
+				StartLine:   eslintMessage.Line,
+				EndLine:     eslintMessage.Line,
+				StartColumn: eslintMessage.Column,
+				EndColumn:   eslintMessage.Column,
+			})
+		}
+	}
+
+	return nil, result
+}

--- a/parsers/eslint_test.go
+++ b/parsers/eslint_test.go
@@ -1,0 +1,40 @@
+package parsers
+
+import (
+	"github.com/cirruslabs/cirrus-ci-annotations/model"
+	"github.com/stretchr/testify/assert"
+	"path/filepath"
+	"testing"
+)
+
+func TestParseESLintAnnotations(t *testing.T) {
+	expected := []model.Annotation{
+		{
+			Level:       model.LevelFailure,
+			Message:     "curly: Expected { after 'if' condition.",
+			Path:        "path/to/file.js",
+			StartLine:   2,
+			EndLine:     2,
+			StartColumn: 1,
+			EndColumn:   1,
+		},
+		{
+			Level:       model.LevelFailure,
+			Message:     "no-process-exit: Don't use process.exit(); throw an error instead.",
+			Path:        "path/to/file.js",
+			StartLine:   3,
+			EndLine:     3,
+			StartColumn: 1,
+			EndColumn:   1,
+		},
+	}
+
+	// Fixture taken from "Working with Custom Formatters"[1] and converted to JSON
+	// [1]: https://eslint.org/docs/developer-guide/working-with-custom-formatters#the-results-object
+	err, actual := ParseESLintAnnotations(filepath.Join("..", "testdata", "eslint.json"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	assert.Equal(t, expected, actual)
+}

--- a/testdata/eslint.json
+++ b/testdata/eslint.json
@@ -1,0 +1,36 @@
+[
+  {
+    "filePath": "path/to/file.js",
+    "messages": [
+      {
+        "ruleId": "curly",
+        "severity": 2,
+        "message": "Expected { after 'if' condition.",
+        "line": 2,
+        "column": 1,
+        "nodeType": "IfStatement"
+      },
+      {
+        "ruleId": "no-process-exit",
+        "severity": 2,
+        "message": "Don't use process.exit(); throw an error instead.",
+        "line": 3,
+        "column": 1,
+        "nodeType": "CallExpression"
+      }
+    ],
+    "errorCount": 2,
+    "warningCount": 0,
+    "fixableErrorCount": 0,
+    "fixableWarningCount": 0,
+    "source": "var err = doStuff();\nif (err) console.log('failed tests: ' + err);\nprocess.exit(1);\n"
+  },
+  {
+    "filePath": "Gruntfile.js",
+    "messages": [],
+    "errorCount": 0,
+    "warningCount": 0,
+    "fixableErrorCount": 0,
+    "fixableWarningCount": 0
+  }
+]

